drpg/sync.py

```diff
@@ -71,25 +72,30 @@ def sync(self) -> None:
     def _process_item(self, product: Product, item: DownloadItem) -> None:
         """Prepare for and download the item to the sync directory."""
 
-        logger.info("Processing: %s - %s", product["products_name"], item["filename"])
+        if not self._dry_run:
+            logger.info("Processing: %s - %s", product["products_name"], item["filename"])
 
-        try:
-            url_data = self._api.file_task(product["products_id"], item["bundle_id"])
-        except self._api.FileTaskException:
-            logger.warning(
-                "Could not download product: %s - %s",
-                product["products_name"],
-                item["filename"],
+            try:
+                url_data = self._api.file_task(product["products_id"], item["bundle_id"])
+            except self._api.FileTaskException:
+                logger.warning(
+                    "Could not download product: %s - %s",
+                    product["products_name"],
+                    item["filename"],
+                )
+                return
+
+            file_response = httpx.get(
+                url_data["download_url"], timeout=30.0, follow_redirects=True
             )
-            return
-
-        file_response = httpx.get(
-            url_data["download_url"], timeout=30.0, follow_redirects=True
-        )
 
         path = self._file_path(product, item)
```

Comment:
Could you please move this line to the top of this method and rewrite the rest to only have a single one check for running in dry mode? Something like:

```python
path = ...
if self._dry_run:
    ...
else:
    ...
```

Reply:
OK, done. Take a look.